src/ng/directive/ngOptions.js

```diff
@@ -467,9 +467,15 @@ var ngOptionsDirective = ['$compile', '$document', '$parse', function($compile,
       if (!multiple) {
 
         selectCtrl.writeValue = function writeNgOptionsValue(value) {
+          var selectedOption = options.selectValueMap[selectElement.val()];
           var option = options.getOptionFromViewValue(value);
 
+          // Make sure to remove the selected attribute from the previously selected option
+          // Otherwise, screen readers might get confused
+          if (selectedOption) selectedOption.element.removeAttribute('selected');
+
           if (option) {
+            if (providedEmptyOption) emptyOption.removeAttr('selected');
```

Comment:
I think ti would be safer to move this inside the `removeEmptyOption()` function, s we don't have to manually call it every time.

For example, there is currently a call for `removeEmptyOption()` inside `selectCtrl.readValue()`. Do we need to also remove the `selected` attribute? Probably not, but who knows :stuck_out_tongue: